src/communication/CommandLine.cpp

```diff
@@ -0,0 +1,475 @@
+                        {
+                            logError("Missing model file with -l argument.");
+                            exit(1);
+                        }
+                        argument = arguments[argument_index];
+
+                        const FMatrix3x3 transformation = last_settings.get<FMatrix3x3>("mesh_rotation_matrix"); //The transformation applied to the model when loaded.
+
+                        if (!loadMeshIntoMeshGroup(&slice.scene.mesh_groups[mesh_group_index], argument.c_str(), transformation, last_extruder.settings))
+                        {
+                            logError("Failed to load model: %s\n", argument.c_str());
+                            exit(1);
+                        }
+                        else
+                        {
+                            last_settings = slice.scene.mesh_groups[mesh_group_index].meshes.back().settings;
+                        }
+                        break;
+                    }
+                    case 'o':
+                        argument_index++;
+                        if (argument_index >= arguments.size())
+                        {
+                            logError("Missing output file with -o argument.");
+                            exit(1);
+                        }
+                        argument = arguments[argument_index];
+                        if (!FffProcessor::getInstance()->setTargetFile(argument.c_str()))
+                        {
+                            logError("Failed to open %s for output.\n", argument.c_str());
+                            exit(1);
+                        }
+                        break;
+                    case 'g':
+                        last_settings = slice.scene.mesh_groups[mesh_group_index].settings;
+                        /* ... falls through ... */
+                    case 's':
+                    {
+                        //Parse the given setting and store it.
+                        argument_index++;
+                        if (argument_index >= arguments.size())
+                        {
+                            logError("Missing setting name and value with -s argument.");
+                            exit(1);
+                        }
+                        argument = arguments[argument_index];
+                        const size_t value_position = argument.find("=");
+                        std::string key = argument.substr(0, value_position);
+                        if (value_position == std::string::npos)
+                        {
+                            logError("Missing value in setting argument: -s %s", argument.c_str());
+                            exit(1);
+                        }
+                        std::string value = argument.substr(value_position + 1);
+                        last_settings.add(key, value);
+                        break;
+                    }
+                    default:
```

Comment:
Suggest to put brackets for more clarity.

Reply:
Done: c79a886a5dda870f325fff8eaf3c9241ef137f33